From 0858da8425a71641fe32222bc91c060856dcb217 Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Thu, 9 Feb 2023 05:55:42 +0900
Subject: [PATCH] nss-myhostname: fix inverted condition in

Fixes a bug introduced by db50d326a46beca3cc24b6354b6e1b3591902d45.

(cherry picked from commit a3b993ca3fb6fc0b837745c1ae82aca684951842)

Resolves: #2167468
---
 src/nss-myhostname/nss-myhostname.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nss-myhostname/nss-myhostname.c b/src/nss-myhostname/nss-myhostname.c
index 120e76be45..93db9d16e7 100644
--- a/src/nss-myhostname/nss-myhostname.c
+++ b/src/nss-myhostname/nss-myhostname.c
@@ -460,7 +460,7 @@ enum nss_status _nss_myhostname_gethostbyaddr2_r(
         } else {
                 assert(af == AF_INET6);
 
-                if (socket_ipv6_is_enabled())
+                if (!socket_ipv6_is_enabled())
                         goto not_found;
 
                 if (memcmp(addr, LOCALADDRESS_IPV6, 16) == 0) {
